student_showcase/api/views.py

```diff
@@ -20,16 +26,81 @@ def api_root(request, format=None):
     'auth/': 'Authenticatication for EDSS API.',
     }, status=200)
 
-class CompanyViewSet(viewsets.ModelViewSet):
-    """
-    RESTful API endpoint for Company
-    """
-    queryset = Company.objects.all()
-    serializer_class = CompanySerializer
-
-    def create(self, request):
-        serializer = CompanyAccountSerializer(data=request.data)
-        serializer.is_valid(raise_exception=True)
-        serializer.save()
-        request.data['password'] = '*' * len(request.data['password'])
-        return JsonResponse({'request': request.data, 'status': 'created successfully'}, status=201)
+class StudentViewSet(viewsets.ModelViewSet):
+    queryset = Student.objects.all()
+    serializer_class = StudentSerializer
+
+    def get_permissions(self):
+        if self.action == 'list':
+            permission_classes = [ListPermissions]
+        elif self.action == 'resumes':
+            permission_classes = [ListPermissions]
+        elif self.action == 'retrieve':
+            permission_classes = [DetailPermissions]
+        elif self.action == 'resume':
+            permission_classes = [DetailPermission]
+        elif self.action == 'update':
+            permission_classes = [AlterPermissions]
+        elif self.action == 'partial_update':
+            permission_classes = [AlterPermissions]
+        elif self.action == 'destroy':
+            permission_classes = [AlterPermissions]
+        else:
+            permission_classes = []
+        return [permission() for permission in permission_classes]
+
+    def update(self, request, pk):
+        try:
+            user = Student.objects.get(id=pk)
+            self.check_object_permissions(request, user)
+        except Student.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+        if "major" in request.data:
+            user.major = request.data['major']
+        if "account" in request.data:
+            user.account = request.data['account']
+        if "major" in request.data:
+            user.major = request.data['major']
+        if "year" in request.data:
+            user.year = request.data['year']
+        if "membership" in request.data:
+            user.membership = request.data['membership']
+        if "clearance" in request.data:
+            user.clearance = request.data['clearance']
+        if "resume" in request.data:
+            user.resume = request.data['resume']
+        if "linked_in" in request.data:
+            user.linked_in = request.data['linked_in']
+        if "attendance" in request.data:
+            user.attendance = request.data['attendance']
+        user.save()
+        return Response(self.serializer_class(user).data, status=201)
+
+    @action(detail=True)
+    def resume(self, request, pk):
+        try:
+            student = Student.objects.get(id=pk)
+            self.check_object_permissions(request, student)
+            print(student)
```

Comment:
remove print statement

Reply:
also fyi, if you haven't been using pdb.set_trace(), I would most definitely recommend.